Update readme (#386)
Arelle/Arelle: README.md

> [![Run Python Unit Tests](https://github.com/Arelle/Arelle/actions/workflows/tests.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/tests.yml)  [![Run Xbrl Conformance Suites](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml)  [![Build and publish python package](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml)  [![Build for Linux](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml)  [![Build for Mac](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml)  [![Build for Windows](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml)

Reviewer: I believe they're all running against the default branch (master), so the status is misleading for most of them. Only the python package and OS builds will update (when we tag a new release). I think we should leave badges out of this PR and consider them in a separate [discussion](#294).

As an aside, maybe we should run all of the test suites on the release workflow (possibly even include the long running EFM suite) before running the builds?

> [![Run Python Unit Tests](https://github.com/Arelle/Arelle/actions/workflows/tests.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/tests.yml)  [![Run Xbrl Conformance Suites](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml)  [![Build and publish python package](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml)  [![Build for Linux](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml)  [![Build for Mac](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml)  [![Build for Windows](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml)

Author: So we are removing all badges for now?

> [![Run Python Unit Tests](https://github.com/Arelle/Arelle/actions/workflows/tests.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/tests.yml)  [![Run Xbrl Conformance Suites](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/conformance-suites.yml)  [![Build and publish python package](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/python-package.yml)  [![Build for Linux](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-linux.yml)  [![Build for Mac](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-mac.yml)  [![Build for Windows](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml/badge.svg)](https://github.com/Arelle/Arelle/actions/workflows/build-windows.yml)

Reviewer: Yes. Looks like Derek just opened a PR for pypi and python versions, but let's remove the rest until we update the release pipelines to also run the test suites.

<div align="center">
  <img src="http://arelle.org/arelle/wp-content/themes/platform/images/logo-platform.png">
</div>

- [Arelle](#arelle)
  - [Features](#features)
- [Installation](#installation)

>   - [Install PyPi package](#install-pypi-package)

Reviewer: nit: the "i" in "PyPI" is capitalized:
```suggestion
  - [Install PyPI package](#install-pypi-package)
```

* QuickBooks is supported by XBRL-GL.

# Installation

> The implementation is in Python 3.10, and is intended for Windows (any recent),
> Mac OS-X 10.9+, Unix or Linux. The standard installation includes both desktop GUI,
> a localhost RESTful web server and CLI.

Reviewer: * We don't officially support Python 3.10 yet. Hopefully within the next week that'll change.
* Preexisting from the installation guide - supported OS versions are out of date

```suggestion
The implementation is in Python 3.9 and is intended for Windows, macOS, or Linux (tested against Ubuntu and RHEL). The standard desktop installation includes a GUI, a RESTful web server, and CLI. We try to support all operating system versions that still receive security updates from their development teams.
```

## Install PyPi package

> ```shell
> pip install arelle-release

Reviewer: We should document [supported optional extras](https://github.com/Arelle/Arelle/blob/bb7eb53e5d26adbd7fb307d9bb4b1c7a0b5b592d/pyproject.toml#L35-L55):
```suggestion
The Arelle python package defines optional extra dependencies for various plugins and use cases.
* Crypto (security plugin dependencies)
* DB (database plugin dependencies)
* EFM (EdgarRenderer plugin dependencies - does not include the EdgarRenderer, just the dependencies required to run it)
* ObjectMaker (ObjectMaker plugin dependencies)
* WebServer (dependencies for running the Arelle web server)
```shell
pip install arelle-release
# or for all extra dependencies
pip install arelle-release[Crypto,DB,EFM,ObjectMaker,WebServer]
```

```

## Install development version from github
```shell

> pip install git+https://git@github.com/arelle/arelle.git@master#egg=arelle-release

Reviewer: I don't think `#egg=arelle-release` is needed?
```suggestion
pip install git+https://git@github.com/arelle/arelle.git@master
```

```

## Install distributions
Distributions include Python version and resources needed to run Arelle.

Arelle provides distributions for the following operating systems:

> * Windows (x64 and x32 architecture)
> * Mac OS
> * Linux (Ubuntu)
> * Red Hat

Reviewer: Preexisting, this should be tweaked:
```suggestion
* Windows
* macOS (Intel)
* Linux (Ubuntu and Red Hat Enterprise Linux)
```

Distributions can be downloaded from:
* [Arelle website](https://arelle.org/arelle/pub/)

> * [github releases](https://github.com/Arelle/Arelle/releases)

Reviewer: nit:
```suggestion
* [GitHub releases](https://github.com/Arelle/Arelle/releases)
```
